launcher/tests/test_updater.py

```diff
@@ -0,0 +1,399 @@
+@mock.patch("Updater.sdlog.error")
+@mock.patch("Updater.sdlog.info")
+def test_check_updates_calls_correct_commands(
+    mocked_info, mocked_error, mocked_call, vm
+):
+    status = updater.check_updates(vm)
+    assert status == UpdateStatus.UPDATES_OK
+    if vm in debian_based_vms:
+        subprocess_call_list = [
+            call(["qvm-run", current_templates[vm], "sudo apt update"]),
+            call(
+                [
+                    "qvm-run",
+                    current_templates[vm],
+                    "[[ $(apt list --upgradable | wc -l) -eq 1 ]]",
+                ]
+            ),
+            call(["qvm-shutdown", current_templates[vm]]),
+        ]
+    elif vm == "fedora":
+        subprocess_call_list = [
+            call(["qvm-run", current_templates[vm], "dnf check-update"]),
+            call(["qvm-shutdown", current_templates[vm]]),
+        ]
+    elif vm == "dom0":
+        subprocess_call_list = [
+            call(["sudo", "qubes-dom0-update", "--check-only"]),
+        ]
+    else:
+        pytest.fail("Unupported VM: {}".format(vm))
+    mocked_call.assert_has_calls(subprocess_call_list)
+    assert not mocked_error.called
+
+
+@mock.patch("Updater.check_updates", return_value=0)
+@mock.patch("subprocess.check_call")
+@mock.patch("Updater.sdlog.error")
+@mock.patch("Updater.sdlog.info")
+def test_check_all_updates(
+    mocked_info, mocked_error, mocked_call, mocked_check_updates
+):
+    updater.check_all_updates()
+    check_updates_call_list = [call(x) for x in current_templates.keys()]
+    mocked_check_updates.assert_has_calls(check_updates_call_list)
+    mocked_subprocess_calls = [
+        call(
+            [
+                "qvm-run",
+                "sd-svs",
+                "echo '0' > {}".format(updater.FLAG_FILE_STATUS_SD_SVS),
+            ]
+        ),
+        call(
+            [
+                "qvm-run",
+                "sd-svs",
+                "echo '{}' > /home/user/sdw-last-updated".format(
+                    str(datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S"))
+                ),
+            ]
+        ),
+    ]
+    mocked_call.assert_has_calls(mocked_subprocess_calls)
+    assert not mocked_error.called
+
+
+@pytest.mark.parametrize("status", UpdateStatus)
+@mock.patch("subprocess.check_call")
+@mock.patch("Updater.sdlog.error")
+@mock.patch("Updater.sdlog.info")
+def test_write_updates_status_flag_sd_svs(
+    mocked_info, mocked_error, mocked_call, status
+):
+    updater._write_updates_status_flag_sd_svs(status)
+    mocked_call.assert_called_once_with(
+        [
+            "qvm-run",
+            "sd-svs",
+            "echo '{}' > {}".format(status.value, updater.FLAG_FILE_STATUS_SD_SVS),
+        ]
+    )
+    assert not mocked_error.called
+
+
+@mock.patch("subprocess.check_call")
+@mock.patch("Updater.sdlog.error")
+@mock.patch("Updater.sdlog.info")
+def test_write_last_updated_flags_to_disk(mocked_info, mocked_error, mocked_call):
```

Comment:
optional: use tmpdir here (and in other tests that will write files to user's homedir)